File: Real-estate-market-IB190099.Services/Database/Message.cs
using System;
using System.Collections.Generic;

namespace Real_estate_market_IB190099.Services.Database;

public partial class Message
{
    public int Id { get; set; }

    public string Content { get; set; } = null!;

    public int? SenderId { get; set; }

    public DateTime? Timestamp { get; set; }

    public int AdvertiseId { get; set; }

    public virtual Advertise Advertise { get; set; } = null!;

    public virtual User? Sender { get; set; }
}

File: Real-estate-market-IB190099.mobile/realestatemobile/lib/model/message.dart
import 'package:realestatemobile/model/user.dart';
// ignore_for_file: non_constant_identifier_names, depend_on_referenced_packages
import 'package:json_annotation/json_annotation.dart';
import 'advertise.dart';
part 'message.g.dart';

@JsonSerializable()
class Message {
  int? id;
  String? content;
  User? sender;
  int? senderId;
  Object? timestamp;
  Advertise? advertise;
  int? advertiseId;
  int? isEmployee;

  Message() {}

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  /// Connect the generated [_$MessageToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$MessageToJson(this);
}
